src/main/scala/cognite/spark/v1/SequenceRowsRelation.scala

```diff
@@ -164,19 +168,36 @@ class SequenceRowsRelation(config: RelationConfig, sequenceId: CogniteId)(val sq
         case _ => throw SparkSchemaHelperRuntime.badRowError(row, "rowNumber", "Long", "")
       }
 
-      val externalId = row.get(externalIdIndex) match {
-        case x: String => x
-        case _ => throw SparkSchemaHelperRuntime.badRowError(row, "externalId", "String", "")
+      val maybeExternalId = Try(row.get(externalIdIndex)).toOption match {
+        case Some(x: String) => Some(x)
+        case _ => None
+      }
+
+      val maybeId = Try(row.get(idIndex)).toOption match {
+        case Some(x: Long) => Some(x)
+        case Some(x: Int) => Some(x.toLong)
+        case _ => None
+      }
+
+      val externalId = (maybeExternalId, maybeId) match {
+        case (Some(externalId), _) => externalId
+        case (None, Some(id)) =>
+          client.sequences.retrieveById(id).unsafeRunSync().externalId match {
```

Comment:
this would fetch the sequence metadata over and over again for each row, we should use the CogniteId union for the two possible cases.

Reply:
RIght, I update the case class and use CogniteId instead ;)